pkg/storage/bucket/client.go

```diff
@@ -119,7 +123,7 @@ type Config struct {
 
 	// Not used internally, meant to allow callers to wrap Buckets
 	// created using this config
-	Middlewares []func(objstore.Bucket) (objstore.Bucket, error) `yaml:"-"`
+	Middlewares []func(objstore.InstrumentedBucket) (objstore.InstrumentedBucket, error) `yaml:"-"`
```

Comment:
Note to reviewers: I will take care to re-vendor Mimir in GEM and fix Middlewares there.